### DISCUSSION TITLE
Configure `publicKey.pem` and `privateKey.pem` to be outside of project

### DISCUSSION THREAD

**Discussion by fenix-hub** [quarkusio/quarkus#29029] [Q&A]
Hi everyone,
I've got a multi-module maven project with several Quarkus microservices.
I'm using a "shared" Quarkus project to let other microservices validate the JWT from incoming requests, using such configuration

*`application.properties` of shared module*
```
### JWT CONFIGURATION
smallrye.jwt.sign.key.location=privatekey.pem
mp.jwt.verify.publickey.location=publickey.pem
```

But this configuration requires each single microservice to have a copy of both private and public key.
Since I will be using docker, is there a way to let the microservices get those keys from a shared folder, outside of the actual project? 
(so the keys will be required to be only in one folder of the server)

Thanks

**Comment by sberyozkin** [comment]
@fenix-hub Hi, right, it gets into the area of the key management, one idea is for this shared project start an endpoint which will return keys and you'd then refer to them using HTTP based URL.
Or if it is acceptable, do have them in the shared folder on the system where the docker is running - which will require accessing this holder folder as volume

**Reply by fenix-hub** [reply]
Hi thanks for you answer.
About the second case, which sounds like the easiest one for now, how would I configure the `.location` property to point to a file in the shared volume?
Notice: I'm compiling uber jars, not native executables.

**Reply by sberyozkin** [reply]
@fenix-hub Something like this I guess,
`docker -v /host-keys:/container-keys` and then refer to `/container-keys/mykey.pem` etc

**Reply by fenix-hub** [reply]
Hi @sberyozkin , I'm back.
It works on docker with no issue whatsoever, the thing is that in this way I'm not able to figure a way to test in _dev_ profile.
I'm using Windows, and even though I've got a `keys/` folder at the root of my project containing my key, I get this error
```
[...]
Caused by: java.net.MalformedURLException: no protocol: ./keys/publickey.pem
````

**Reply by sberyozkin** [reply]
@fenix-hub Hi, I'm not sure about the dev profile, can you type the steps you follow with the devmode here please, my understanding  the dev profile applies to the locally defined properties
